docs/source/ucx-debug.rst

```diff
@@ -0,0 +1,117 @@
+    | Message size: 100000000                                                                  |
+    +------------------------------------------------------------------------------------------+
+                10     0.000  9104.800  9104.800   10474.41   10474.41         110         110
+
+
+``-c`` option is NUMA dependent and sets the CPU Affinity of process for a partiular GPU.  CPU Affinity information can be found in ``nvidia-smi topo -m``
+::
+
+    user@mlnx:~$  nvidia-smi topo -m
+            GPU0    GPU1    GPU2    GPU3    GPU4    GPU5    GPU6    GPU7    mlx5_0  mlx5_1  mlx5_2  mlx5_3  CPU Affinity
+    GPU0     X      NV1     NV1     NV2     NV2     SYS     SYS     SYS     PIX     PHB     SYS     SYS     0-19,40-59
+    GPU1    NV1      X      NV2     NV1     SYS     NV2     SYS     SYS     PIX     PHB     SYS     SYS     0-19,40-59
+    GPU2    NV1     NV2      X      NV2     SYS     SYS     NV1     SYS     PHB     PIX     SYS     SYS     0-19,40-59
+    GPU3    NV2     NV1     NV2      X      SYS     SYS     SYS     NV1     PHB     PIX     SYS     SYS     0-19,40-59
+    GPU4    NV2     SYS     SYS     SYS      X      NV1     NV1     NV2     SYS     SYS     PIX     PHB     20-39,60-79
+    GPU5    SYS     NV2     SYS     SYS     NV1      X      NV2     NV1     SYS     SYS     PIX     PHB     20-39,60-79
+    GPU6    SYS     SYS     NV1     SYS     NV1     NV2      X      NV2     SYS     SYS     PHB     PIX     20-39,60-79
+    GPU7    SYS     SYS     SYS     NV1     NV2     NV1     NV2      X      SYS     SYS     PHB     PIX     20-39,60-79
+    mlx5_0  PIX     PIX     PHB     PHB     SYS     SYS     SYS     SYS      X      PHB     SYS     SYS
+    mlx5_1  PHB     PHB     PIX     PIX     SYS     SYS     SYS     SYS     PHB      X      SYS     SYS
+    mlx5_2  SYS     SYS     SYS     SYS     PIX     PIX     PHB     PHB     SYS     SYS      X      PHB
+    mlx5_3  SYS     SYS     SYS     SYS     PHB     PHB     PIX     PIX     SYS     SYS     PHB      X
+
+    Legend:
+
+      X    = Self
+      SYS  = Connection traversing PCIe as well as the SMP interconnect between NUMA nodes (e.g., QPI/UPI)
+      NODE = Connection traversing PCIe as well as the interconnect between PCIe Host Bridges within a NUMA node
+      PHB  = Connection traversing PCIe as well as a PCIe Host Bridge (typically the CPU)
+      PXB  = Connection traversing multiple PCIe bridges (without traversing the PCIe Host Bridge)
+      PIX  = Connection traversing at most a single PCIe bridge
+      NV#  = Connection traversing a bonded set of # NVLinks
+
+NVLink
+------
+
+System Configuration
+~~~~~~~~~~~~~~~~~~~~
+
+Again, using the ``nvidia-smi topo -m`` command which show the hardware topology of the GPUs and display NVLink information.  For example, using the output above, we can see that GPUs 0-4 are conneted to another with NVLink,
+and GPUs 5-8 are also connected with NVLink.  However, these sets of GPUs are only connected to one another with InfiniBand.
```

Comment:
The description you wrote for NVLink is incorrect, please refer to https://images.nvidia.com/content/pdf/dgx1-v100-system-architecture-whitepaper.pdf (PDF page 11/document page 9). We can't say some GPUs are connected to "one" NVLink and other GPUs are connected to "another" NVLink, and connectivity is not homogenous. What happens is we have GPUs connected that are connected by a single NVLink connection (NV1), others with two NVLink connections (NV2), and some not connected at all via NVLink (SYS).

Can you please rewrite that? My suggestion would be something like:

"Note that NVLink connectivity on the system above (DGX-1) is not homogenous, some GPUs are connected by a single NVLink connection (NV1, e.g., GPUs 0 and 1), others with two NVLink connections (NV2, e.g., GPUs 1 and 2), and some not connected at all via NVLink (SYS, e.g., GPUs 3 and 4)."

Reply:
Thanks Peter!